frontend/lib/resolution/Resolver.cpp

```diff
@@ -1502,21 +1502,54 @@ void Resolver::issueErrorForFailedModuleDot(const Dot* dot,
 
 }
 
-void Resolver::handleResolvedCall(ResolvedExpression& r,
-                                  const uast::AstNode* astForErr,
-                                  const CallInfo& ci,
-                                  const CallResolutionResult& c) {
+bool Resolver::handleResolvedCallWithoutError(ResolvedExpression& r,
+                                              const uast::AstNode* astForErr,
+                                              const CallInfo& ci,
+                                              const CallResolutionResult& c) {
 
   if (!c.exprType().hasTypePtr()) {
-    issueErrorForFailedCallResolution(astForErr, ci, c);
     r.setType(QualifiedType(r.type().kind(), ErroneousType::get(context)));
+    return true;
   } else {
     r.setMostSpecific(c.mostSpecific());
     r.setPoiScope(c.poiInfo().poiScope());
     r.setType(c.exprType());
     // gather the poi scopes used when resolving the call
     poiInfo.accumulate(c.poiInfo());
   }
+  return false;
+}
+
+void Resolver::handleResolvedCall(ResolvedExpression& r,
+                                  const uast::AstNode* astForErr,
+                                  const CallInfo& ci,
+                                  const CallResolutionResult& c) {
+
+  if (handleResolvedCallWithoutError(r, astForErr, ci, c)) {
+    issueErrorForFailedCallResolution(astForErr, ci, c);
+  }
+}
+
+void Resolver::handleResolvedCallPrintCandidates(ResolvedExpression& r,
+                                                 const uast::Call* call,
+                                                 const CallInfo& ci,
+                                                 const Scope* scope,
+                                                 const PoiScope* poiScope,
+                                                 const QualifiedType& receiverType,
+                                                 const CallResolutionResult& c) {
+
+  if (handleResolvedCallWithoutError(r, call, ci, c)) {
+    std::vector<ApplicabilityResult> rejected;
+    std::ignore = resolveCallInMethod(context, call, ci, scope, poiScope,
```

Comment:
This matches the way that our "multiple overloads of variable" error works -- by default, resolution doesn't do tracing/tracking of failed candidates. If it fails, however, it's re-run while gathering additional information (in the scope resolution, the additional info is trace of searched scopes; in this PR, the additional info is the list of rejected candidates).

The reason for this is to avoid consuming memory and resources for the "good" case (i.e., we don't want to list 100 failed `operator +` candidates each time we _successfully_ resolve `x+y`).